canvas_sdk/commands/base.py

```diff
@@ -1,33 +1,73 @@
 import json
 import re
 from enum import EnumType
-from typing import get_args
+from typing import Any, Literal, get_args
 
-from pydantic import BaseModel, ConfigDict, model_validator
-from typing_extensions import Self
+from pydantic import BaseModel, ConfigDict
+from pydantic_core import InitErrorDetails, PydanticCustomError, ValidationError
 
 from canvas_sdk.effects import Effect, EffectType
 
 
 class _BaseCommand(BaseModel):
-    model_config = ConfigDict(strict=True, validate_assignment=True)
+    model_config = ConfigDict(strict=True, revalidate_instances="always")
 
     class Meta:
         key = ""
+        originate_required_fields = (
+            "user_id",
+            "note_uuid",
+        )
+        edit_required_fields = (
+            "user_id",
+            "command_uuid",
+        )
+        delete_required_fields = (
+            "user_id",
+            "command_uuid",
+        )
+        commit_required_fields = (
+            "user_id",
+            "command_uuid",
+        )
+        enter_in_error_required_fields = (
+            "user_id",
+            "command_uuid",
+        )
 
```

Comment:
I can see us using this pattern of setting required fields and validation on actions for more than just commands. For example, I was planning on integrating it into [my work on Tasks](https://github.com/canvas-medical/canvas-plugins/pull/64). Should we put this into a base class that is more general than `_BaseCommand`?

cc: @aduane

Reply:
love that idea! i can create a separate validation class for it